project.go

```diff
@@ -64,3 +65,28 @@ func (p *Project) MakeParams() gps.SolveParameters {
 
 	return params
 }
+
+// BackupVendor looks for existing vendor directory and if it's not empty,
+// creates a backup of it to a new directory with the provided suffix.
+func BackupVendor(vpath, suffix string) (string, error) {
+	// Check if there's a non-empty vendor directory
+	vendorExists, err := IsNonEmptyDir(vpath)
+	if err != nil {
+		return "", err
+	}
+	if vendorExists {
+		vendorbak := vpath + "-" + suffix
+		// Check if a directory with same name exists
+		if _, err = os.Stat(vendorbak); os.IsNotExist(err) {
+			// Rename existing vendor to vendor-{suffix}
+			if err := renameWithFallback(vpath, vendorbak); err != nil {
+				return "", err
+			}
+			return vendorbak, nil
+		} else {
```

Comment:
Sorry, that's not what I meant.

I meant removing the `else` part and still have the `return "", errVendorBackupFailed`.